### ISSUE TITLE
Make radius and theta params work on all marks

### ISSUE THREAD

**Issue by nyurik** [vega/vega#1425]
Text mark has two special params - `radius` and `theta` that allow mark positioning to shift in radial coordinates relative to `x, y`.  I think these two parameters are useful for all marks, and not just text. For example, drawing symbol marks on top of a pie chart would not require me to do trigonometry calculations to place it into a specific pie chart sector.

**Comment by jheer** [comment]
Thanks! I think this makes sense for `symbol` marks, which have an unambiguous point location. Probably for `image` marks, too, handled similarly to `text` marks (with placement relative to a defined anchor point).

I'm not sure this makes sense for *all* other marks though. For example, `arc` marks are already radial, and really are polar coordinate analogues of `rect` marks. For `line` and `area` marks, similar polar variants might make more sense as a future enhancement.

Also, for marks with extents (`rect`, `rule`) there is the question of how to resole  `x` / `x2` with `radius` / `radius2`, etc. So at minimum we'd need a more detailed proposal to consider changes there.

**Comment by nyurik** [comment]
Makes sense. I guess at least for the initial release, rect and similar make no sense. Line would be interesting, as it would allow easy way to create spiral/spiderweb still graphs.
